spec/core/query_spec.rb

```diff
@@ -334,4 +334,14 @@
       end
     end
   end
+
+  describe "apply_where" do
+    context "matches regexp with block input" do
+      subject { dv.apply_where(dv.match /weeks/) { |x| "#{x.split.first.to_i * 7} days" } }
```

Comment:
@zverok - I remember seeing DSL support being added to `which` in PR #396. Should a similar DSL call be supported here, like `apply_which`?